app.rb

```diff
@@ -31,38 +33,46 @@ def set_locale(locale)
     else
       redirect '/'
     end
-    @board = @game.current_state.each_slice(3).to_a
+    @board = @game.current_state.each_slice(@game.board.row_size).to_a
     erb :game
   end
 
   post '/game' do
     if !params[:game_type] || !params[:first_player]
-      redirect '/'
-    end
-
-    case params[:first_player]
-    when "first_player"
-      first_player = true
-      second_player = false
+        @error = t "error", get_locale 
+        erb :welcome
     else
-      first_player = false
-      second_player = true
-    end
-
-    case params[:game_type]
-    when "HvH"
-      session[:game] = GameStateManager.new(Human.new("X", first_player), Human.new("Y", second_player))
-    when "HvC"
-      session[:game] = GameStateManager.new(Human.new("X", first_player), Computer.new("Y", second_player))
+        case params[:first_player]
+        when "first_player"
+          first_player = true
+          second_player = false
+        else
+          first_player = false
+          second_player = true
+        end
+
+        case params[:board_size]
```

Comment:
you shouldn't need a case statement if there are only two options. and also lets not use magic strings. https://code.tutsplus.com/tutorials/refactoring-legacy-code-part-2-magic-strings-constants--cms-20527. this could be refactored to
`if params[:board_size] == FOUR_BY_FOUR`

Reply:
Roger that, thx for the link